src/Executor/AbstractExecutor.php

```diff
@@ -76,23 +76,19 @@ public function getPurger()
     /**
      * Set the logger callable to execute with the log() method.
      *
-     * @param callable $logger
-     *
      * @return void
      */
-    public function setLogger($logger)
+    public function setLogger(callable $logger)
```

Comment:
Why is the logger a `callable` anyway? 🙈

Reply:
Line 94. 
```php
        $logger = $this->logger;
        $logger($message);
```

You think we should change it to accept a `Psr\Log\LoggerInterface`?

Reply:
We should look into that, yes. But probably in a separate PR.

Reply:
I'm not going to do that, because it's used for console output directly.

https://github.com/doctrine/DoctrineFixturesBundle/blob/c808a0c85c38c8ee265cc8405b456c1d2b38567d/Command/LoadDataFixturesDoctrineCommand.php#L137-L139

Reply:
> it's used for console output directly.

We could try to leverage `Symfony\Component\Console\Logger\ConsoleLogger`. My problem is that a closure is a bad logging abstraction. While it works for the way the bundle uses it, it sucks if you want the logs to go anywhere else, e.g. you want to use a logging framework instead.

Deprecation: #462